### PR TITLE
mrc 5087: Add endpoint to list parameters on a report

### PR DESCRIPTION
**Pull request by r-ash**
Note this uses https://github.com/mrc-ide/orderly2/pull/129 and should be merged after that

This PR will add an endpoint to list the parameters available on a report for a particular name and hash

Builds on #3 so should be merged after that

### CODE REVIEW

File: R/reports.R [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  exprs <- parse(text = contents)
+  orderly2::orderly_parse(exprs = exprs, filename = basename(path))$parameters
+}
+
+
+get_orderly_script_path <- function(name, ref, root) {
+  contents <- gert::git_ls(root, ref = ref)
+  re <- sprintf("^src/%s/(%s|orderly)\\.R$", name, name)
+  matches <- grep(re, contents$path, value = TRUE, perl = TRUE)
+  if (length(matches) > 1) {
```
Comment (M-Kusumgar) [inline R/reports.R]: should we care about the case where they have none? or assumed to have at least one at this point?

**Review by M-Kusumgar — APPROVED** [review]
Code looks good, see tiny comment, approving pending that orderly2 change which might change the function name depending on the way you choose to do it!

File: R/reports.R (hunk 1 above)
Comment (r-ash) [reply R/reports.R]: API should only call this for directories which have an `orderly.R` script. So this shouldn't be triggered with matches 0. That said, better to not rely on that knowledge as the error message raised if there is no file is quite ugly. So I've expanded this check to test if `!= 0`

File: DESCRIPTION [hunk 2]
```diff
@@ -32,7 +32,7 @@ Suggests:
     testthat (>= 3.0.0)
 Config/testthat/edition: 3
 Remotes:
-    mrc-ide/orderly2,
+    mrc-ide/orderly2@add-ast-read,
```
Comment (richfitz) [inline DESCRIPTION]: This is ready to be merged now

```suggestion
    mrc-ide/orderly2,
```

File: DESCRIPTION [hunk 3]
```diff
@@ -18,7 +18,7 @@ Imports:
     gert (>= 2.0.1),
     ids,
     jsonlite,
-    orderly2,
+    orderly2 (>= 1.99.12),
```
Comment (richfitz) [inline DESCRIPTION]:
```suggestion
    orderly2 (>= 1.99.13),
```

**Review by richfitz — APPROVED** [review]
LGTM with the https://github.com/mrc-ide/orderly2/pull/129 merged and DESCRIPTION here updated

**Pull request merged**